server-api/grpc/map.proto

```diff
@@ -0,0 +1,352 @@
+syntax = "proto3";
+
+package monai.deploy.platform;
+
+option csharp_namespace = "Monai.Deploy.Platform.Grpc";
+option go_package = "monaid.grpc";
+option java_package = "com.monai.deploy.platform.grpc";
+
+import public "common.proto";
+
+// Service for managing MONAI Application Packages (MAP).
+service Map {
+  // Requests the creation of a new MAP registation.
+  // Attempting to create a MAP with an already known container URN will result in an error.
+  rpc Create(MapCreateRequest) returns (MapCreateResponse);
+
+  // Requests the deletion of a known MAP registation.
+  // Identifying the MAP registration to delete by reference name will also delete the reference name.
+  // Attempting to delete a MAP registration with reference names will result in an error.
+  // Attempting to delete an unknown or already deleted MAP registation will result in an error.
+  rpc Delete(MapDeleteRequest) returns (MapDeleteResponse);
+
+  // Requests a filtered list of know MAP records.
+  // When no filter is provided, the default filter will be used.
+  rpc List(MapListRequest) returns (stream MapListResponse);
+
+  // Requests the deletion of a MAP reference name.
+  rpc ReferenceDelete(MapReferenceDeleteRequest) returns (MapReferenceDeleteResponse);
+
+  // Requests for a reference name to refer to known MAP.
+  // Attempting to reference a deleted or error-state MAP will result in an error.
+  rpc ReferenceUpdate(MapReferenceUpdateRequest) returns (MapReferenceUpdateResponse);
+
+  // Requests for a deleted MAP registration to restored (undeleted).
+  // Restoring a MAP registration does restore any deleted reference names.
+  // Attempting to restore a MAP registration which is not deleted with result in an error.
+  // Attempting to restore an unknown MAP registration will result in an error.
+  rpc Restore(MapRestoreRequest) returns (MapRestoreResponse);
```

Comment:
What's the scenario imagined her? Is it because of an accidental delete?  I definitely think there is need for some kind of rollback/restore workflow, could this fulfil that?

Reply:
I agree